manager/integration/tests/test_scheduling.py

```diff
@@ -754,16 +754,15 @@ def test_data_locality_basic(client, core_api, volume_name, pod, settings_reset)
 
     volume3 = client.by_id_volume(volume3_name)
     assert len(volume3.replicas) == 1
-    volume3 = client.by_id_volume(volume3_name)
 
     create_and_wait_pod(core_api, pod3)
     wait_for_rebuild_start(client, volume3_name)
     volume3 = client.by_id_volume(volume3_name)
     assert len(volume3.replicas) == 2
     wait_for_rebuild_complete(client, volume3_name)
 
-    volume3 = client.by_id_volume(volume3_name)
-    assert len(volume3.replicas) == 1
+    # Wait for deletion of extra replica
+    volume3 = wait_for_volume_replica_count(client, volume3_name, 1)
```

Comment:
In case of the rebuilding of a replica due to data locality, we need to wait for deletion of the extra replica before counting the number of replicas